### PR TITLE
refactor(date): replace i18n-js with i18next

### PR DESCRIPTION
**Pull request by mmaciorowski-sage**
### Proposed behaviour
Use i18next for the date component
https://github.com/Sage/carbon/blob/master/rfcs/text/i18n.md

### Current behaviour
i18n-js

### Checklist

- [x] Commits follow our style guide
- [ ] Screenshots are included in the PR if useful
- [ ] All themes are supported if required
- [x] Unit tests added or updated if required
- [x] Cypress automation tests added or updated if required
- [x] Storybook added or updated if required
- [x] Typescript `d.ts` file added or updated if required
- [ ] Carbon implementation and Design System documentation are congruent

### Testing instructions
https://codesandbox.io/s/carbon-quickstart-forked-du7dk?file=/src/index.js

No test changes are required.
`date-picker`, `date-range` and `date` components should be tested for regression.

### CODE REVIEW

File: src/__experimental__/components/date/date.stories.js [hunk 1]
```diff
@@ -18,6 +19,15 @@ export default {
   },
 };
 
+i18n.changeLanguage("en-US");
+i18n.addResourceBundle("en-US", "carbon", {
+  date: {
+    formats: {
+      javascript: "M/D/YYYY",
```
Comment (ianforster-sage) [inline src/__experimental__/components/date/date.stories.js:26]: Has this short date format been browser tested? Sometimes a date without a leading zero can throw an error. It's a different format to that which you've added to the helper for the same `us` locale, nor is it in the `_defaultDateFormats` list.

Comment (mmaciorowski-sage) [reply src/__experimental__/components/date/date.stories.js:26]: yes it works well, but probably I should remove it from a story.
Zrzut ekranu 2021-03-5 o 15 24 11

File: src/utils/helpers/date/__spec__.js [hunk 2]
```diff
@@ -1,18 +1,26 @@
...
-    I18n.translations = {
-      en: {
-        date: { formats: { inputs: ["DD/MM/YYYY"] } },
-      },
-
-      us: {
-        date: { formats: { inputs: ["MM/DD/YYYY"] } },
-      },
-    };
+    console.log(i18n.language);
```
Comment (nicktitchmarsh) [inline src/utils/helpers/date/__spec__.js:23]: Is there supposed to be a console log in here?

Comment (mmaciorowski-sage) [reply src/utils/helpers/date/__spec__.js]: good point, removed.

**Review by nicktitchmarsh — APPROVED** [review]

**Review by ianforster-sage — APPROVED** [review]

**Review by Alec-Walker-Sage — APPROVED** [review]

**Pull request merged**